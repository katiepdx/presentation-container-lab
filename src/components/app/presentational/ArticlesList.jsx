// Display a list of <Article/> components
// Needs a snapshot test

import React, { Component } from 'react'
import Loading from '../presentational/Loading'
import { getNewsArticles } from '../services/news-api'

export default class ArticlesList extends Component {
  // set state 
  state = {
    loading: true,
    userSearch: 'bitcoin',
    foundArticles: []
  }

  // when page loads, fetch from API, update state, and change loading to false
  async componentDidMount() {
    // fetch from api using user's search
    const foundArticles = await getNewsArticles();
  
    // set state 
    this.setState({ foundArticles: foundArticles, loading: false})
  }

  render() {
    // deconstruct state 
    const { loading, foundArticles } = this.state;

    console.log(foundArticles, 'ARTICLES LIST IN STATE IN RENDER')
    // conditional rendering 
    if(loading) return <Loading />

    return (
      <div>
        not loading
      </div>
    )
  }
}
