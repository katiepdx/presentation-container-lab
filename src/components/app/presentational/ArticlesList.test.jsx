import React from 'react'
import { render, cleanup } from '@testing-library/react'
import ArticlesList from './ArticlesList'

describe('ArticlesList Component', () => {
  afterEach(() => cleanup());

  it('renders ArticlesList', () => {
    // snapshot fragment and render ArticlesList (pass test data array props)
    const { asFragment } = render(<ArticlesList
        foundArticles={[
          {
            title: 'Article Title 1',
            author: 'Author 1',
            description: 'Description 1',
            key: 'Article Title 1'
          },{
            title: 'Article Title 2',
            author: 'Author 2',
            description: 'Description 2',
            key: 'Article Title 2'
          },{
            title: 'Article Title 3',
            author: 'Author 3',
            description: 'Description 3',
            key: 'Article Title 3'
          }
        ]}
      />);
      // expect that the snapshot matches the asFragment
      expect(asFragment()).toMatchSnapshot();
  })
})
