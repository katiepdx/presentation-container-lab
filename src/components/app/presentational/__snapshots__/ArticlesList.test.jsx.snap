// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList Component renders ArticlesList 1`] = `
<DocumentFragment>
  <article>
    <div>
      <h2>
        Article Title 1
      </h2>
      <h6>
        Author 1
      </h6>
      <p>
        Description 1
      </p>
    </div>
  </article>
  <article>
    <div>
      <h2>
        Article Title 2
      </h2>
      <h6>
        Author 2
      </h6>
      <p>
        Description 2
      </p>
    </div>
  </article>
  <article>
    <div>
      <h2>
        Article Title 3
      </h2>
      <h6>
        Author 3
      </h6>
      <p>
        Description 3
      </p>
    </div>
  </article>
</DocumentFragment>
`;
